diff --git a/src/pc/proxy.h b/src/pc/proxy.h
--- a/src/pc/proxy.h	(revision 847fe7905954f3ae883de2936415ff567aa9039b)
+++ b/src/pc/proxy.h	(date 1757588036708)
@@ -64,11 +64,14 @@
 #include "rtc_base/event.h"
 #include "rtc_base/string_utils.h"  // IWYU pragma: keep
 #include "rtc_base/thread.h"
-#include "rtc_base/trace_event.h"
 
 #if !defined(RTC_DISABLE_PROXY_TRACE_EVENTS) && !defined(WEBRTC_CHROMIUM_BUILD)
 #define RTC_DISABLE_PROXY_TRACE_EVENTS
 #endif
+
+#if !defined(RTC_DISABLE_PROXY_TRACE_EVENTS)
+#include "rtc_base/trace_event.h"
+#endif
 
 namespace webrtc {
 
